app/src/main/java/tech/ula/utils/ValidationUtility.kt

```diff
@@ -1,31 +1,86 @@
 package tech.ula.utils
 
+import android.content.res.Resources
+import tech.ula.R
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 
-class ValidationUtility() {
+class ValidationUtility(systemResources: Resources) {
 
-    fun isUsernameValid(password: String): Boolean {
-        val pattern: Pattern
-        val matcher: Matcher
+    private val resources = systemResources
 
-        val USERNAME_REGEX = "^[a-zA-Z0-9]*\$"
+    fun validateUsername(username: String): Credential {
+        var usernameIsValid = false
+        var errorMessage = ""
+        val blacklist = resources.getStringArray(R.array.blacklisted_usernames)
 
-        pattern = Pattern.compile(USERNAME_REGEX)
-        matcher = pattern.matcher(password)
+        when {
+            username.isEmpty() -> {
+                errorMessage = resources.getString(R.string.error_empty_field)
+            }
+            !validateUsernameCharacters(username) -> {
+                errorMessage = resources.getString(R.string.error_username_invalid_characters)
+            }
+            blacklist.contains(username) -> {
+                errorMessage = resources.getString(R.string.error_username_in_blacklist)
+                errorMessage = String.format(errorMessage, username)
+            }
+            else ->
+                usernameIsValid = true
+        }
 
-        return matcher.matches()
+        return Credential(usernameIsValid, errorMessage)
     }
 
-    fun isPasswordValid(password: String): Boolean {
+    fun validatePasswords(password: String, vncPassword: String): Credential {
+        var passwordIsValid = false
+        var errorMessage = ""
+
+        when {
```

Comment:
Same trick could be employed with this `when`.

Reply:
Changed it here too.